src/util/xml_utils.js

```diff
@@ -31,22 +31,24 @@ const XML_HEADER = '<?xml version="1.0" encoding="UTF-8"?>';
  *   <root> <a>1</a> <b>2</b> <a>3</a> <b>4</b> <z>42</z> <c d="5"></c> </root>
  *
  */
-function encode_xml(object, ignore_header) {
+function encode_xml(object, ignore_header, repeat_array_tags) {
     let output = ignore_header ? '' : XML_HEADER;
     append_object(s => {
         output += s;
-    }, object);
+    }, object, repeat_array_tags);
     return output;
 }
 
-function append_object(append, object) {
+function append_object(append, object, repeat_array_tags, father) {
     if (typeof(object) !== 'object') {
         append(encode_xml_str(object));
     } else if (Array.isArray(object)) {
         // arrays are encoded without adding new tags
         // which allows repeating keys with the same name
         for (let i = 0; i < object.length; ++i) {
-            append_object(append, object[i]);
+            if (repeat_array_tags && i !== 0) append('<' + father + '>');
+            append_object(append, object[i], repeat_array_tags, father);
+            if (repeat_array_tags && object.length !== i + 1) append('</' + father + '>');
```

Comment:
this does not support tag attributes, for this encoder mode I would change the level of recursion and encode the arrays tags by the caller